cmd/kubeadm/app/componentconfigs/configset.go

```diff
@@ -97,40 +98,77 @@ func (h *handler) FromCluster(clientset clientset.Interface, clusterCfg *kubeadm
 	return h.fromCluster(h, clientset, clusterCfg)
 }
 
+// known holds the known component config handlers. Add new component configs here.
+var known = []*handler{
```

Comment:
Just for confirmation.
This approach works for well-known component configs only; whenever addons will start to add unknown component config to the picture, an equivalent solution should be provided

Reply:
That's quite unlikely. The idea is to have here the component configs of things, that are managed by kubeadm. The configs of things managed by the addon manager can be piped directly to it without ever going through this module.
In fact configs (such as the kube-proxy one), can be optionally excluded at some point from this list if an addon manager is used.